apiserver/facades/client/secrets/secrets.go

```diff
@@ -275,7 +284,11 @@ func (s *SecretsAPI) secretContentFromBackend(ctx context.Context, uri *coresecr
 			return nil, errors.NotFoundf("external secret backend %q, have %q", backendID, s.backends)
 		}
 		val, err = backend.GetContent(ctx, ref.RevisionID)
-		if err == nil || !errors.Is(err, errors.NotFound) || lastBackendID == backendID {
+		notFound := errors.Is(err, secreterrors.SecretNotFound) || errors.Is(err, secreterrors.SecretRevisionNotFound)
+		if err == nil || !notFound || lastBackendID == backendID {
+			if notFound {
```

Comment:
Move out from the if.

Reply:
Can't because we can still get to here if `lastBackendID == backendID`

Reply:
So you want to ignore the not found if lastBackendID != backendID.

Reply:
No, we still return the error but just not this user friendly one for not found (line 292)